### PR TITLE
Add time context to telemetry requests

### PR DESCRIPTION
**Pull request by scottbell**
Closes [#139](https://github.com/akhenry/openmct-yamcs/issues/139)

### Describe your changes:
Added a time context for telemetry requests, so telemetry providers can see if a request is in real-time or fixed-time.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/api/time/TimeContext.js [hunk 1]
```diff
@@ -362,6 +362,19 @@ class TimeContext extends EventEmitter {
         this.boundsVal = newBounds;
         this.emit('bounds', this.boundsVal, true);
     }
+
+    /**
+     * Checks if this time context is in real-time mode or not.
+     * @returns {boolean} true if this context is in real-time mode, false if not
+     * @private
```
Comment (ozyx) [inline src/api/time/TimeContext.js:369]: Should this be marked as private?

File: src/api/time/TimeContext.js [hunk 2]
```diff
@@ -362,6 +362,19 @@ class TimeContext extends EventEmitter {
...
+    */
+    isRealtime() {
+        if (this.clock()) {
```
Comment (ozyx) [inline src/api/time/TimeContext.js:372]: should we add a quick test in `TimeAPISpec` to cover this new method?

File: src/api/telemetry/TelemetryAPI.js [hunk 3]
```diff
@@ -131,8 +130,9 @@ export default class TelemetryAPI {
 
     /**
      * @private
+     * Though also used by TelemetryCollection
```
Comment (ozyx) [inline src/api/telemetry/TelemetryAPI.js:133]: In this case, should we remove the `@private`?

**Review by ozyx — CHANGES_REQUESTED** [review]
Looking good! Smoke tested and works well. Have just a few small suggestions

File: src/api/time/TimeContext.js (hunk 2 above)
Comment (scottbell) [reply src/api/time/TimeContext.js:372]: The e2e test in the YAMCS plugin is testing this implicitly, but I agree a test in the core openmct is a good idea.

File: src/api/telemetry/TelemetryAPI.js (hunk 3 above)
Comment (scottbell) [reply src/api/telemetry/TelemetryAPI.js:133]: We probably should 😀, I was just reluctant as @akhenry specifically [wanted it private](https://github.com/akhenry/openmct-yamcs/issues/139#issuecomment-1283271861).

File: src/api/time/TimeContext.js (hunk 1 above)
Comment (scottbell) [reply src/api/time/TimeContext.js:369]: Absolutely not!

**Review by ozyx — APPROVED** [review]
LGTM!

**Pull request merged**